### PR TITLE
added auth.teams.list method

### LINKED ISSUE #1137: web-api is missing auth.teams.list
### Description

The web-api package is missing a binding for https://api.slack.com/methods/auth.teams.list

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [x] bug
- [x] enhancement (feature request)

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

#### Packages:

- [x] `@slack/web-api`

### PR DESCRIPTION
**Pull request by stevengill**
###  Summary

Added `auth.teams.list` method. Fixes #1137 

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

**Review by mwbrooks — APPROVED** [review]
Nice! 🚀

**Comment by seratch** [comment]
LGTM but we can verify the builds after merging #1129

**Review by seratch — APPROVED** [review]

**Pull request merged**
